### PR TITLE
feat(Application-system): Custom delegation support.

### PR DESCRIPTION
**Pull request by Thorrik58**
# Custom delegation support.

Attach a link to issue if relevant

## What

Adding in custom delegation support for application system.
If a user has a custom delegation and wants to access an application that allows custom delegations. The user must have the required scope for that application defined in the template in order to do so.
First test will be the veiðileyfi application.

## Why

Needed to reduce workload on procuration holders and allow 3rd parties to submit applications for the company when needed.

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by thordurhhh — APPROVED** [review]
Code owned files lgtm

File: apps/application-system/api/src/app/modules/application/application.controller.ts
Comment (obmagnusson) [inline apps/application-system/api/src/app/modules/application/application.controller.ts:198]: Should we permit the client to determine whether a scope check should occur? Is it possible for a user to elevate their access privileges by modifying the query?

File: apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts
Comment (obmagnusson) [inline apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts:168]: We should consider moving this API call to the client library, creating a separate service for better reusability.

**Review by obmagnusson — COMMENTED** [review]
Some initial thoughts

File: apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts
Comment (Thorrik58) [reply apps/application-system/api/src/app/modules/application/tools/applicationAccess.service.ts:168]: I agree, would be happy to implement it once this has been merged to main as there is some pressure to get this in before the freeze.

File: apps/application-system/api/src/app/modules/application/application.controller.ts
Comment (Thorrik58) [reply apps/application-system/api/src/app/modules/application/application.controller.ts:198]: I think its ok as its just filtering out unneeded data, the client can already see his scopes if he wishes to.
But @eirikurn would probably be the best to determine if its a bigger issue.

**Review by obmagnusson — APPROVED** [review]

File: libs/clients/auth/public-api/src/lib/authPublicApiClient.module.ts
Comment (eirikurn) [inline libs/clients/auth/public-api/src/lib/authPublicApiClient.module.ts:14]: Why?

Services should be able to inject it if it's been loaded correctly in the application root module.

File: libs/auth-api-lib/src/lib/delegations/delegations-incoming.service.ts
Comment (eirikurn) [inline libs/auth-api-lib/src/lib/delegations/delegations-incoming.service.ts:200]: I would really like to merge scopes (and types) across delegations in a smarter way to support interesting edge/use cases, but it's not a priority for now.

**Review by eirikurn — APPROVED** [review]
Nice work. Mainly concerned about the performance of the findAll endpoint.

Approving to unblock. Feel free to prioritise the different performance suggestions in this or later PRs.

File: apps/application-system/api/src/app/modules/application/application.controller.ts
Comment (Thorrik58) [reply apps/application-system/api/src/app/modules/application/application.controller.ts:238]: Good catch, should've thought about that. Ill fix what I can.

File: libs/clients/auth/public-api/src/lib/authPublicApiClient.module.ts
Comment (Thorrik58) [reply libs/clients/auth/public-api/src/lib/authPublicApiClient.module.ts:14]: I made the change to get my schemas to build.
No idea why it behaves like it does on my end.
```
[App] Error	7/18/2023, 5:45:50 PM [ExceptionHandler] Nest can't resolve dependencies of the AuthPublicApiClientConfiguration (?). Please make sure that the argument CONFIGURATION(AuthPubliClient) at index [0] is available in the AuthPublicApiClientModule context.

       Potential solutions:
       - If CONFIGURATION(AuthPubliClient) is a provider, is it part of the current AuthPublicApiClientModule?
       - If CONFIGURATION(AuthPubliClient) is exported from a separate @Module, is that module imported within AuthPublicApiClientModule?
         @Module({
           imports: [ /* the Module containing CONFIGURATION(AuthPubliClient) */ ]
         })
```

**Pull request merged**
